mathics/core/expression.py

```diff
@@ -317,7 +317,7 @@ def __repr__(self) -> str:
 
     def __str__(self) -> str:
         return "%s[%s]" % (
-            str(self.head),
+            self.get_head_name(short=True),
```

Comment:
This could be a problem if there are symbols with the same name in different contexts, which is something that you would want to detect in debugging.

Reply:
`repr` still shows the long names.  

Personally in my debugging, I don't recall needing this yet.  And in WMA I am seeing short names used in errors and tracing. 

We could add a user-defined setting, but this is more complicated.

Reply:
This is useful and related, but different.  

This is a useful Mathics3 user function.  But in addition, there is always the need for lower-level displays of elements. These appear in error messages as well as in that TraceEvaluation.

What I've been noticing is that WMA does not fill out context paths when reporting error messages and we do. I can't find a good example of this but I am sure you've seen experienced these.

Reply:
Yep, because uses "FullForm".

Reply:
Ok - then this can wait until after we have boxing under control.